weave

```diff
@@ -1382,14 +1382,28 @@ case "$COMMAND" in
     attach)
         collect_cidr_args "$@"
         shift $CIDR_ARG_COUNT
-        if [ "$1" = "--or-die" ]; then
-          ATTACH_TYPE="_or_die"
-          shift
-        fi
+        while [ $# -gt 0 ]; do
+            case "$1" in
+                --or-die)
+                    ATTACH_TYPE="_or_die"
+                    ;;
+                --rewrite-hosts)
+                    REWRITE_HOSTS=1
+                    ;;
+                *)
+                    break
+                    ;;
+            esac
+            shift
+        done
         [ $# -eq 1 ] || usage
         CONTAINER=$(container_id $1)
         create_bridge
         ipam_cidrs$ATTACH_TYPE $CONTAINER $CIDR_ARGS
+        if [ -n "$REWRITE_HOSTS" ] && command_exists weavehosts ; then
+            PATH_AND_NAME=$(docker inspect -f '{{.HostsPath}} {{.Config.Hostname}}' $CONTAINER)
+            weavehosts $PATH_AND_NAME $ALL_CIDRS
```

Comment:
You get the container ID echoed to `stdout` and the message "Container [long_hex_number] not running."  This message comes from the `with_container_netns` on the next line, i.e. these additions have not changed anything.

Docker does not mind you inspecting or writing the filesystem of a container that has exited.  Aha!, you say, what if you add `--rm`, but that is only possible when running via the proxy, in which case the container would still be in `weavewait` at this point.